File: Minimum Absolute Difference in Sliding Submatrix/Minimum Absolute Difference in Sliding Submatrix.py
# Solution for Minimum Absolute Difference in Sliding Submatrix in PY

class Solution:
    def minAbsDiff(self, grid, k):
        m, n = len(grid), len(grid[0])

        result = [[0] * (n - k + 1) for _ in range(m - k + 1)]

        for rowStart in range(m - k + 1):
            for colStart in range(n - k + 1):

                submatrix_elements = []

                for r in range(rowStart, rowStart + k):
                    for c in range(colStart, colStart + k):
                        submatrix_elements.append(grid[r][c])

                submatrix_elements.sort()

                min_diff = float('inf')

                for i in range(1, len(submatrix_elements)):
                    if submatrix_elements[i] != submatrix_elements[i - 1]:
                        min_diff = min(min_diff,
                                       submatrix_elements[i] - submatrix_elements[i - 1])

                result[rowStart][colStart] = 0 if min_diff == float('inf') else min_diff

        return result

File: Minimum Absolute Difference in Sliding Submatrix/Minimum Absolute Difference in Sliding Submatrix.java
// Solution for Minimum Absolute Difference in Sliding Submatrix in JAVA

class Solution {
    public int[][] minAbsDiff(int[][] grid, int k) {
        int m = grid.length;
        int n = grid[0].length;

        int[][] result = new int[m - k + 1][n - k + 1];

        for (int rowStart = 0; rowStart <= m - k; rowStart++) {
            for (int colStart = 0; colStart <= n - k; colStart++) {

                List<Integer> submatrixElements = new ArrayList<>();

                for (int r = rowStart; r < rowStart + k; r++) {
                    for (int c = colStart; c < colStart + k; c++) {
                        submatrixElements.add(grid[r][c]);
                    }
                }

                Collections.sort(submatrixElements);

                int minDiff = Integer.MAX_VALUE;

                for (int i = 1; i < submatrixElements.size(); i++) {
                    if (!submatrixElements.get(i).equals(submatrixElements.get(i - 1))) {
                        minDiff = Math.min(minDiff,
                                submatrixElements.get(i) - submatrixElements.get(i - 1));
                    }
                }

                result[rowStart][colStart] = (minDiff == Integer.MAX_VALUE ? 0 : minDiff);
            }
        }

        return result;
    }
}
